src/luxonis_ml/data/utils/data_utils.py

```diff
@@ -18,52 +18,131 @@ def get_granule(filepath, addition, component_name):
     return granule
 
 
-def check_classification(val1, val2):
+def assert_classification_format(dataset, val):
+    if val is not None:
+        if isinstance(val, str):
+            if val not in dataset.fo_dataset.classes.get("class", []):
+                raise ClassNotFoundError(f"Class {val} is not found in dataset")
```

Comment:
Maybe unknown class?

Reply:
"ClassUnknownError" for example. Even better might be if we stick with "Exception"?